docs/DeveloperGuide.md

```diff
@@ -1,38 +1,337 @@
+<br><br><br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 2. Product Scope
+<br>
 
+### 2.1. Target User Profile
 
-## Product scope
-### Target user profile
+- Undergraduate Students
+- Comfortable using CLI applications
+- Able to type relatively quickly
+<br>
 
-{Describe the target user profile}
+### 2.2. Value proposition
+This application seeks to help the target users to keep track of and manage their module components and deadlines, as it can be confusing to juggle so many deliverables at once.
+<br><br><br>
 
-### Value proposition
+## 3. About this developer guide
+<br>
 
-{Describe the value proposition: what problem does it solve?}
+### 3.1. Purpose
+This developer guide aims to allow you to understand the design and implementation considerations for Mod Happy. With this guide, you will be able to add on or modify any existing implementation for your own usage.
+<br>
 
-## User Stories
+### 3.2. Explanation of notation
+`Text` formatted as such represent the classes and functions implemented in the application, and user input examples. 
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+> 📔 <span style="color:#00bb00">**NOTE:**</span>
+>
+> Text enclosed in this "Note" block should be taken note of as it can contain additional important information about the Component/Implementation.
 
-## Non-Functional Requirements
+<br><br><br>
 
-{Give non-functional requirements}
+## 4. Acknowledgements
 
-## Glossary
+- Some foundational source code was adapted from [addressbook-level2](https://github.com/se-edu/addressbook-level2).
+- Google's [GSON library](https://github.com/google/gson) was used to facilitate serialisation and deserialisation of data stored in the data file.
 
-* *glossary item* - Definition
+<br><br><br>
 
-## Instructions for manual testing
+## 5. Design
+
+The following architecture diagram provides a high level overview of the main components of Mod Happy and how they interact with one another.
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Components.puml)
+
+The `Main` class is responsible for handling program initialisation, termination, as well as the application's main execution logic.
+
+Mod Happy's components are summarised below:
+
+* `UI`: Manages the application's text UI.
+* `Parser`: Interprets user input and returns corresponding `Command` objects.
+* `Command`: Handles command execution logic.
+* `Data`: Manages module and task data in program memory.
+* `Storage`: Reads data from, and writes data to Mod Happy's data storage files.
+
```

Comment:
This is a really clear explanation and diagram of the entire design of the program that's easily understood!